--- a/cores/arduino/USB/USBCore.cpp
+++ b/cores/arduino/USB/USBCore.cpp
@@ -261,7 +261,8 @@
 			utox8(SERIAL_NUMBER_WORD_1, &name[8]);
 			utox8(SERIAL_NUMBER_WORD_2, &name[16]);
 			utox8(SERIAL_NUMBER_WORD_3, &name[24]);
-			name[32] = '\0';
+			static const char suffix[] = "-8MU";
+			memcpy(&name[32], suffix, sizeof(suffix));
 			return sendStringDescriptor((uint8_t*)name, setup.wLength);
 #endif
 		}
